### PR TITLE
Core: FP-723 Fix blocking form submitting on ajax timeout

### PR DESCRIPTION
**Pull request by olgagiza-home24**
[FP-723]

Added error ajax handling for remote validation
* timeouts marks field as valid
* 5xx http codes marks field as valid
* 4xx http codes marks field as invalid

[FP-723]: https://home24.atlassian.net/browse/FP-723

### CODE REVIEW

**Review by asantangelo — APPROVED** [review]

File: test/methods.js [hunk 1]
```diff
@@ -294,6 +294,166 @@ asyncTest("remote", function() {
...
+	});
+	strictEqual( v.element(e), false);
+	e.val("Peter");
+	strictEqual( v.element(e), true);
+});
+
+asyncTest("remote failure", function() {
+	expect(3);
+
+	var e = $("#username"),
```
Comment (robertfabian-home24) [inline test/methods.js:332]: Maybe you could use more explicit variable names than the one-letter-long ones?

**Review by robertfabian-home24 — APPROVED** [review]
Apart from one small comment, LGTM

File: test/methods.js (hunk 1 above)
Comment (olgagiza-home24) [reply test/methods.js:332]: It looks like that's convention used in those tests, if you'll take a look at the other tests all elements are called `e` and validators are called `v`

Comment (robertfabian-home24) [reply test/methods.js:332]: That's a shitty convention :P but ok, let's not overcomplicate this review and keep things consistent.

Comment (olgagiza-home24) [reply test/methods.js:332]: Noone from us chose that - it's a fork of https://github.com/jquery-validation/jquery-validation/ XD

**Pull request merged**